pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -209,6 +209,8 @@ self: super: {
   ### END HASKELL-LANGUAGE-SERVER SECTION ###
   ###########################################
 
+  basement = appendPatch ./patches/basement-fix-32-bit.patch super.basement;
```

Comment:
A couple things:

- We generally try not to carry around patches in Nixpkgs.  Is there any way you could pull this patch from upstream using `fetchpatch`?  There should be a bunch of examples in this file.
- I'm a little bit concerned about taking patches for one of these foundational packages that upstream hasn't approved.  Could you make it so this patch is only applied on the architectures it is needed, instead of every architecture?
- Could you add a comment in the code explaining what this patch is for, when it can be removed, where it came from, etc?

Reply:
>We generally try not to carry around patches in Nixpkgs. Is there any way you could pull this patch from upstream using fetchpatch? There should be a bunch of examples in this file.

Yes, but it's tricky to do because technically the upstream patch is itself broken (it doesn't cover enough GHC versions.) We could do some string replacement here, but it's non-obvious - it would involve replacing one compiler version with another, only on added lines in the patch.

>Could you make it so this patch is only applied on the architectures it is needed, instead of every architecture?

This is done now (in fact the patch broke other architectures.)

>Could you add a comment in the code explaining what this patch is for, when it can be removed, where it came from, etc?

Will do.